src/iOS/Xamarin.iOS/ViewControllers/SampleInfoViewController.cs

```diff
@@ -0,0 +1,245 @@
+﻿using ArcGISRuntime.Samples.Shared.Models;
+using Foundation;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using UIKit;
+
+namespace ArcGISRuntime
+{
+    public class SampleInfoViewController : UIViewController
+    {
+        // SampleInfo for the sample provided by the constructor.
+        private SampleInfo _info;
+
+        // Control for switching between readme and source code viewing.
+        private UISegmentedControl _switcherControl;
+
+        // Web view for viewing readme markdown.
+        private UIWebView _readmeView;
+
+        // Controls for the source code viewer.
+        private UIView _codeView;
+        private UIWebView _codeWebView;
+        private UIBarButtonItem _codeButton;
+
+        // Dictionary where keys are filenames and values are HTML of source code.
+        private Dictionary<string, string> _sourceCodeFiles = new Dictionary<string, string>();
+
+        // Directory for loading HTML locally.
+        private string _contentDirectoryPath = Path.Combine(NSBundle.MainBundle.BundlePath, "Content/");
+
+        public SampleInfoViewController(SampleInfo info, UISegmentedControl switcher)
+        {
+            _info = info;
+            _switcherControl = switcher;
+        }
+
+        public override void ViewDidLoad()
+        {
+            base.ViewDidLoad();
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            // Build out readme html
+            try
+            {
+                string readmePath = Path.Combine(NSBundle.MainBundle.BundlePath, "Samples", _info.Category, _info.FormalName, "readme.md");
+                string readmeCSSPath = Path.Combine(NSBundle.MainBundle.BundlePath, "SyntaxHighlighting/github-markdown.css");
+                string readmeContent = new MarkedNet.Marked().Parse(File.ReadAllText(readmePath));
+
+                string readmeHTML = "<!doctype html><head><base href=\"" +
+                    readmePath +
+                    "\"><link rel=\"stylesheet\" href=\"" +
+                    readmeCSSPath +
+                    "\" /></head><body class=\"markdown-body\">" +
+                    readmeContent +
+                    "</body>";
+
+                // Load the readme into the webview.
+                _readmeView.LoadHtmlString(readmeHTML, new NSUrl(_contentDirectoryPath, true));
```

Comment:
Note: currently, when clicking readme links, they open directly in the about viewer; I recommend overriding the URL load behavior and launching an external browser when the user clicks links in the about page.